plugins/ingest/src/main/java/org/elasticsearch/ingest/Data.java

```diff
@@ -129,4 +129,33 @@ public String getId() {
     public boolean isModified() {
         return modified;
     }
+
+    public Map<String, Object> asMap() {
```

Comment:
I think how we parse in general needs to be given another look, so we can update that in another PR, what do you think? I thought this would be a sufficient and low-impact placeholder until we figure out how to include metadata into `Data`... and change `Data`'s name to something else :)